hooks/shotgun_launch_shotgun_app.py

```diff
@@ -26,57 +26,51 @@
 
 HookBaseClass = sgtk.get_hook_baseclass()
 
-class LaunchAssociatedApp(HookBaseClass):
-    def execute(self, path, context, associated_entity, **kwargs):
+
+class LaunchShotgunApp(HookBaseClass):
+    def execute(self, published_file, **kwargs):
         """
         Launches the associated app and starts tank.
-        
-        :param path: full path to the published file
-        :param context: context object representing the publish
-        :param associated_entity: same as context.entity
-        """
-        status = False
-
-        if context is None:
-            raise TankError("Context cannot be None!")
 
+        :param published_file: The published file to launch.
+        :raises: `TankError` if no valid application was found.
+        """
         ########################################################################
         # Example implementation below:
+        path = self.get_publish_path(published_file)
 
+        if published_file.get("task"):
+            context = self.tank.context_from_entity("Task", published_file["task"].get("id"))
+        else:
+            context = self.tank.context_from_path(path)
+        if context is None:
```

Comment:
I'm not sure this is even possible? You're in a Project no matter what right? So the context would at least have a Project...

Reply:
I have to take care of this in an extra ticket, but right now the context can be None:
no Task entity linked to the Published File, no filesystem configured for the project.
I'm gonna add a fallback on "entity" instead of "sg_task", and on "project" if "entity" is not provided.